File: domain/src/main/java/com/thinkzi/weather/domain/usecase/base/SynchronousUseCase.java
package com.thinkzi.weather.domain.usecase.base;

/**
 * provide common execution interface of all synchronous UseCase(Clean Architecture)
 * */
public interface SynchronousUseCase<Rs, Ps> {

    interface Callback<Rs> {

        void onSuccess(Rs _return);

        void onError(Throwable _throwable);

    }

    void execute(Ps _parameter, Callback<Rs> _callback);

}

File: domain/src/main/java/com/thinkzi/weather/domain/usecase/base/SynchronousUseCase.kt
package com.thinkzi.weather.domain.usecase.base

/**
 * provide common execution interface of all synchronous UseCase(Clean Architecture)
 */
interface SynchronousUseCase<Rs, Ps> {

    interface Callback<Rs> {

        fun onSuccess(_return: Rs)

        fun onError(_throwable: Throwable)

    }

    fun execute(_parameter: Ps, _callback: Callback<Rs>)

}
